src/dispatch/incident/service.py

```diff
@@ -25,40 +24,58 @@
 SECONDS_IN_HOUR = 3600
 
 
-def resolve_incident_commander_email(
+def assign_incident_role(
     db_session: SessionLocal,
+    incident: Incident,
     reporter_email: str,
-    incident_type: str,
-    incident_name: str,
-    incident_title: str,
-    incident_description: str,
-    page_commander: bool,
+    role: ParticipantRoleType,
 ):
-    """Resolves the correct incident commander email based on given parameters."""
-    # We resolve the incident commander email
+    """Assigns incident roles."""
+    # We resolve the incident role email
     # default to reporter if we don't have an oncall plugin enabled
+    assignee_email = None
```

Comment:
If you assign `assignee_email = reporter_email` here you could avoid all the assignments below by checking if the service exists vs not.